rootex/framework/entity_factory.cpp

```diff
@@ -223,10 +227,15 @@ void EntityFactory::setupLiveEntities()
 {
 	for (auto& entity : m_Entities)
 	{
-		entity.second->setupEntities();
+		setupEntity(entity.second);
 	}
 }
 
+void EntityFactory::setupEntity(Ref<Entity> entity)
+{
+	entity->setupEntities();
+}
+
```

Comment:
This is not needed. You can revert this because it adds a function which only is used once